File: src/main/javabak/tile_entities/capabilities/SimpleOutputFluidStore.java
package redcrafter07.processed.block.tile_entities.capabilities;

import net.minecraft.core.NonNullList;
import net.neoforged.neoforge.fluids.FluidStack;

public class SimpleOutputFluidStore extends SimpleFluidStore {

    public SimpleOutputFluidStore(NonNullList<FluidStack> tanks, int capacity) {
        super(tanks, capacity);
    }

    public SimpleOutputFluidStore(int tanks, int capacity) {
        super(tanks, capacity);
    }

    @Override
    public int fill(FluidStack stack, FluidAction action) {
        return stack.getAmount();
    }

    @Override
    public boolean isFluidValid(int tank, FluidStack stack) {
        return false;
    }

}

File: src/main/kotlin/redcrafter07/processed/block/tile_entities/capabilities/SimpleOutputFluidStore.kt
package redcrafter07.processed.block.tile_entities.capabilities

import net.minecraft.core.NonNullList
import net.neoforged.neoforge.fluids.FluidStack
import net.neoforged.neoforge.fluids.capability.IFluidHandler.FluidAction

class SimpleOutputFluidStore(tanks: NonNullList<FluidStack>, capacity: Int) : SimpleFluidStore(tanks, capacity) {
    constructor(tanks: Int, capacity: Int) : this(NonNullList.withSize(tanks, FluidStack.EMPTY), capacity)

    override fun fill(stack: FluidStack, action: FluidAction): Int = stack.amount
    override fun isFluidValid(tank: Int, stack: FluidStack): Boolean = false
}